jobs/tools/ICON_to_point.py

```diff
@@ -0,0 +1,362 @@
+    '''
+    Get the 2 nearest distances between ICON grid points and specified
+    station altitudes
+
+    Parameters
+    ----------
+    model_topography : 1D np.array
+        This is the elevation over mean sea level of the ICON grid
+    
+    model_levels : 2D np.array
+        Dimensions [ICON_heights, number_of_samples]
+
+    station_ground_elevation : list or 1D np.array
+        e.g., [20,] or np.array([72,180,40])
+        This is the elevation of the *ground over mean sea level*
+
+    station_altitude_over_ground : list or 1D np.array
+        e.g., [15,] or np.array([15, 21, 42])
+        This is the altitude of the *station above the ground*
+    
+    interpolation_strategy : list of strings
+        e.g., ['ground',] or ['ground','mountain','ground']
+        Can be 'ground' or 'mountain', or 'middle' (the latter is between the ground and mountain approach)
+        'ground' uses the model topography + station altitude over ground
+        'mountain' uses the absolute altitude over mean sea level
+
+    Returns
+    -------
+    vertical_distances : 3D np.array
+        Contains the absolute (!) distance between the target point(s)
+        and its 2 nearest neighbour levels
+
+    vertical_indices: 3D np.array
+        Contains the indices to the ICON height levels of the corresponding 2
+        nearest neighbour levels
+    '''
+    # Get the target sampling altitude with a list comprehension
+    target_altitude = [
+        model_topography.isel({
+            "station": i
+        }).values +
+        station_altitude_over_ground[i] if strategy == 'ground' else
+        np.repeat(station_ground_elevation[i], model_topography.shape[1]) +
+        station_altitude_over_ground[i] if strategy == 'mountain' else
+        np.repeat(station_ground_elevation[i], model_topography.shape[1]) / 2 +
+        model_topography.isel({
+            "station": i
+        }).values / 2 + station_altitude_over_ground[i]
+        # if strategy=='middle'
+        for (i, strategy) in enumerate(interpolation_strategy)
+    ]
+    target_altitude = xr.DataArray(target_altitude, dims=['station', 'ncells'])
+
+    # Select 2 closest neighbouring levels
+    first_negative = (model_levels <= target_altitude).argmax(
+        dim=model_levels.dims[0])  # First index where model lies below target
+    vertical_indices = np.stack(
+        [first_negative, first_negative - 1],
+        axis=0)  # Second index thus lies /above/ the target
+    vertical_indices[:, first_negative == 0] = model_levels.values.shape[
+        0] - 1  # If no result found: sample lies below lowest model level. Set it to the lowest model level
+
+    # Sample the corresponding vertical distances between the target and the model levels
+    vertical_distances = np.take_along_axis(
+        (model_levels - target_altitude).values, vertical_indices, axis=0)
+
+    return np.abs(vertical_distances).T, vertical_indices.T
+
+
+def ICON_to_point(longitude,
```

Comment:
Please use lowercase for function names. I would also suggest du replace `ICON` with `icon` in variable names.
```suggestion
def icon_to_point(longitude,
```

Reply:
OK, done!